shared-models/command.js

```diff
@@ -15,23 +47,22 @@ class CommandModel {
      * @param {boolean} [settings.bypassSpamDetection = false] Set to true to not penalize the user for spamming the command.
      * @param {string} [settings.scope = "everywhere"] Scope where the command can be triggered, can be "everywhere|pm|guild"
      */
-    constructor(settings) {
-        const options = Object.assign({}, CommandModel.defaultSettings, settings);
-
-        this._triggerType = options.triggerType;
-        this._trigger = options.trigger;
-        this._help = options.help;
+    constructor({ triggerType = "command", trigger = [], help = "", args = [], ownersOnly = false, scope = "everywhere", requiresMention = false, bypassSpamDetection = false, action = () => { } }) {
+        this._triggerType = triggerType;
+        this._trigger = trigger;
+        this._help = help;
         this._args = [];
-        for (let i = 0; i < options.args.length; ++i) {
-            const arg = Object.assign({}, CommandModel.defaultArgSettings, options.args);
-            this._args.push(arg);
+
+        for (let i = 0; i < args.length; ++i) {
+            this._args.push(new CommandArgumentModel(args[i]));
```

Comment:
Why isn't args of type `CommandArgumentModel[]` instead?

Reply:
I did not want to expose the CommandArgumentModel class. Changing this means moving the "complexity" of creating a CommandArgument in the Plugin class, instead of Command. I would probably keep it that way?